Zusammenführung beide Sensoren (MH-Z19C & DHT11)

NeroCyer opened the pull request:
Zusammenführung beide Sensoren (MH-Z19C & DHT11),
Ausgabe der Temperatur (°C), Luftfeuchtigkeit(%)
und der Luftqualität(ppm).

NeroCyer committed:
Zusammenführung beide Sensoren (MH-Z19C & DHT11),
Ausgabe der Temperatur (°C), Luftfeuchtigkeit(%)
und der Luftqualität(ppm).

janbellenberg commented on arduino/arduino.ino:
```diff
@@ -0,0 +1,160 @@
+    return;
+  }
+  
+  int ppm, temperature = 0;
```
ppm hat hier noch keinen Wert

janbellenberg commented on arduino/arduino.ino:
```diff
@@ -0,0 +1,160 @@
+    digitalWrite(LEDyellow, LOW);
+    digitalWrite(LEDred, LOW);
+  }
+  else if ((ppm < 1000 && (hudty < 40 || hudty > 60) && temp < 26) || (ppm < 1000 && (40 > hudty && hudty < 60 ) && temp > 25) || ((ppm > 1000 && ppm < 1500) && (40 > hudty && hudty < 60 ) && temp < 26))
```
Bitte bei den langen if-Abfragen Zeilenumbrüche einfügen

janbellenberg commented on arduino/arduino.ino:
```diff
@@ -0,0 +1,160 @@
+  // zurückgibt.
+  
+  memset(response, 0, 9);
+  while (co2Serial.available() == 0)
```
Überprüf mal, ob es auch mit !co2Serial.available() geht

janbellenberg commented on arduino/arduino.ino:
```diff
@@ -0,0 +1,160 @@
+  memset(response, 0, 9);
+  while (co2Serial.available() == 0)
+  {
+    delay(1000);
```
Muss die Zeit so lange sein?

NeroCyer replied:
![Delay_Sensor_DHT11](https://user-images.githubusercontent.com/85667519/122389616-681e4380-cf71-11eb-80cc-bcc95fb9555d.jpg)

janbellenberg commented on arduino/arduino.ino:
```diff
@@ -0,0 +1,160 @@
+#define LEDred 8
+
+SoftwareSerial co2Serial(2, 3); // define MH-Z19 RX TX
+DHT dht11HNBK(DHTPIN, DHTTYPE);
```
Warum HNBK? 😂

NeroCyer replied:
Why not? ;D

janbellenberg reviewed (changes requested):
Ganz gut, nur ein paar Sachen zu verbessern

NeroCyer committed:
Verbesserungen durchgeführt...

janbellenberg commented on arduino/arduino.ino:
```diff
@@ -0,0 +1,165 @@
+  int ppm = 0, temperature = 0;
+  readSensor(&ppm, &temperature);
+  
+  if (ppm < 1000 && (40 > hudty && hudty < 60 ) && temp < 26) // gute Luft
```
Da ist bei hudty noch ein Fehler in der Abfrage

NeroCyer replied:
Danke dir für den Hinweis, ich gehe der Sache gerne nach :D

NeroCyer committed:
Update arduino.ino

EN:
Requested fix on if-statement at line 39: the humidity was declared as under 40 instead of above...

DE:
Gewünschte Korrektur der if-Anweisung in Zeile 39: die Luftfeuchtigkeit wurde als unter 40 statt über definiert...

janbellenberg commented:
Wird geschlossen
